controllers/front/notify.php

```diff
@@ -273,8 +279,8 @@ public function prepareVariables()
     public function getMerchantOrderId()
     {
         try {
-            $table = _DB_PREFIX_ .self::ORDERS_TABLE;
-            $sql = 'select ps_order_id from ' . $table .' where id = ' .(int)$this->merchantCartId;
+            $sql = 'select ps_order_id from ' . _DB_PREFIX_ .self::ORDERS_TABLE .' where id = '
+                .(int)$this->merchantCartId . ' and token = \'' . $this->token . '\'';
```

Comment:
scaping token parameter? why not id? Remove both?

Reply:
I'm preserving simple quote for string sql parameter not scaping